File: go-client-sdk/pkg/models/shared/eventattendee.go
package shared

type EventAttendeeAssociations struct {
	AssociationID string `json:"associationId"`
	EventID       string `json:"eventId"`
}

// EventAttendee
// Event Attendees hold information about someone who attendeed or was invited to an event. Attendees are always associated with some Event and another person object such as a Contact, Lead, or Other.
type EventAttendee struct {
	Additional           map[string]interface{}    `json:"additional,omitempty"`
	AssociatedObjectType string                    `json:"associatedObjectType"`
	Associations         EventAttendeeAssociations `json:"associations"`
	CreatedTime          string                    `json:"createdTime"`
	Email                *string                   `json:"email,omitempty"`
	ID                   string                    `json:"id"`
	ModifiedTime         string                    `json:"modifiedTime"`
	NativeID             string                    `json:"nativeId"`
	Status               *string                   `json:"status,omitempty"`
}

File: typescript-client-sdk/src/sdk/models/shared/eventattendee.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";



export class EventAttendeeAssociations extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=associationId" })
  associationId: string;

  @SpeakeasyMetadata({ data: "json, name=eventId" })
  eventId: string;
}


// EventAttendee
/** 
 * Event Attendees hold information about someone who attendeed or was invited to an event. Attendees are always associated with some Event and another person object such as a Contact, Lead, or Other.
**/
export class EventAttendee extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=additional" })
  additional?: Record<string, any>;

  @SpeakeasyMetadata({ data: "json, name=associatedObjectType" })
  associatedObjectType: string;

  @SpeakeasyMetadata({ data: "json, name=associations" })
  associations: EventAttendeeAssociations;

  @SpeakeasyMetadata({ data: "json, name=createdTime" })
  createdTime: string;

  @SpeakeasyMetadata({ data: "json, name=email" })
  email?: string;

  @SpeakeasyMetadata({ data: "json, name=id" })
  id: string;

  @SpeakeasyMetadata({ data: "json, name=modifiedTime" })
  modifiedTime: string;

  @SpeakeasyMetadata({ data: "json, name=nativeId" })
  nativeId: string;

  @SpeakeasyMetadata({ data: "json, name=status" })
  status?: string;
}
